FLT/CrazyFrobenius.lean

```diff
@@ -0,0 +1,986 @@
+  mul_smul _ _ := by
+     intro h
+     show Ideal.comap _ _ = _
+     simp
+     exact rfl
+    -- 'exact rfl' worked, because the two sides of the goal were definitionally equal.
+
+-- The following lemma will be used in `CRTRepresentative` (which is why we
+-- `@[simp]` it), where it is combined with the assumption (implicit in Milne),
+-- that `Q` is a *nonzero* prime ideal of `B`.
+-- it is placed here by convention, as a property of `gal_action_Ideal`.
+-- Jujian helped me to write this lemma.
+@[simp] lemma gal_action_Ideal'.smul_bot (σ : (L ≃ₐ[K] L)) : σ • (⊥ : Ideal' A K L B) = ⊥ := by
+  change Ideal.comap (AlgEquiv.symm (galRestrict A K L B σ)) ⊥ = ⊥
+  apply Ideal.comap_bot_of_injective
+  exact AlgEquiv.injective (AlgEquiv.symm ((galRestrict A K L B) σ))
+
+-- I define the decomposition group of `(Ideal' A K L B)` over `K`
+-- to be the stabilizer of the MulAction `gal_action_Ideal'`.
+
+variable {A K L B} in
+/-- The decomposition group of `Q : Ideal' A K L B` over `K`.
+This is a subgroup of `L ≃ₐ[K] L`. -/
+def decomposition_subgroup_Ideal' (Q : Ideal' A K L B) :
+  Subgroup (L ≃ₐ[K] L) := MulAction.stabilizer (L ≃ₐ[K] L) Q
+
+-- The following variables introduce instances of the residue fields `(A ⧸ P)` and
+-- `(B ⧸ Q)` as finite fields.
+variable (P : Ideal A) [Ideal.IsMaximal P] [Fintype (A ⧸ P)] (P_ne_bot : P ≠ ⊥)
+  (Q : Ideal' A K L B) [Ideal.IsMaximal Q] [Fintype (B ⧸ Q)] (Q_ne_bot : Q ≠ (⊥ : Ideal B))
+  [Algebra (A ⧸ P) (B ⧸ Q)] [IsScalarTower A (A ⧸ P) (B ⧸ Q)] -- last variable suggested by Amelia
+
+/-- `Fintype.card (A ⧸ P)` -/
+local notation "q" => Fintype.card (A ⧸ P)
+-- The main existence proof for the Frobenius element (not accomplished here)
+-- will demonstrate that the order of the Frobenius element is this `q`.
+
+-- By the following instance, we obtain that there is an element `g` in
+-- `(B ⧸ Q)ˣ` which generates `(B ⧸ Q)ˣ`, as a cyclic group.
+-- We will use the existence of such a `g' to prove the main theorem, `Generator'.
+variable {A K L B} in
+instance residue_field_units_is_cyclic : IsCyclic (B ⧸ Q)ˣ :=
+  isCyclic_of_subgroup_isDomain (Units.coeHom _) <| by
+    unfold Function.Injective
+    simp_all
+    intros a b
+    apply Units.ext_iff.2
+
+/-- The Orbit-Stabilizer Theorem for the orbit of the non-zero prime ideal `Q` of `B`
+under the action of the Galois group `L ≃ₐ[K] L`. -/
+noncomputable def galOrbitStabilizer : (MulAction.orbit  (L ≃ₐ[K] L) Q) ≃
+    (L ≃ₐ[K] L) ⧸ ( decomposition_subgroup_Ideal' Q) := by
+  exact MulAction.orbitEquivQuotientStabilizer (L ≃ₐ[K] L) Q
+
+namespace CRTRepresentative.auxiliary
+
+variable {A K L B} in
+-- Amelia helped me to write the following definition.
+/-- Function assigning a representative non-zero prime of type `(Ideal' A K L B)`
+to an element in the orbit of `Q` under the action of the Galois group `L ≃ₐ[K] L`.
+We will consider the orbit to be a 'Fintype', as used in the statement of
+the Chinese Remainder Theorem. -/
+noncomputable def f :
```

Comment:
This is a special case of https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/GroupAction/Quotient.html#MulAction.ofQuotientStabilizer